### ISSUE TITLE
Add-MyTaskCategory should not delete default categories

### ISSUE THREAD

**Issue by jdhitsolutions** [jdhitsolutions/MyTasks#26]
this command should not clear the built-in defaults. Let the user use Remove-MyTaskCategory to delete.

**Comment by jdhitsolutions** [comment]
Or I might need to re-visit how I am storing these values.

**Comment by shaneis** [comment]
This "got" me today cause I didn't realise that the defaults were deleted when a custom one was added.
I don't count this as a big deal, it's more unintuitive more than anything else.

I'll aim to look into this if you have no objection? 😄

**Comment by jdhitsolutions** [comment]
Feel free to take a go at it. I've re-opened this issue.

**Comment by shaneis** [comment]
Writing down my understanding of this in case it's wrong or there's a got'cha that I haven't spotted yet.

## Current understanding

The first time that a category is added, a check is done to see if the MyCategory.txt file exists. If it doesn't we create a empty file.

#### MyTasksFunctions.psm1: Add-MyTaskCategory
```powershell
if (-Not (Test-Path -Path $global:myTaskCategory)) {
    Write-Verbose [...]
    Set-Content -Value "" -Path $global:myTaskCategory -Encoding Unicode
}
```

Otherwise, we grab the contents and add the new categories that aren't in there yet.

#### MyTasksFunctions.psm1: Add-MyTaskCategory
```powershell
        #get current contents
        $current = Get-Content -Path $global:myTaskCategory -Encoding Unicode | where-object {$_ -match "\w+"}
    } #begin

    Process {
        foreach ($item in $Category) {
            if ($current -contains $($item.trim())) {
                [Check for duplicates]
            }
            else {
                Write-Verbose [...]
                Add-Content -Value $item.Trim() -Path $global:myTaskCategory -Encoding Unicode
            }
        }
    } #process
```

## Potential Change

If we make a change to the `MyTasks.psm1` file to make the `$myTaskDefaultCategories` variable be module scoped e.g.

```powershell
$script:myTaskDefaultCategories = "Work", "Personal", "Other", "Customer"
```

Then we could change the `Add-MyTaskCategory` to add the default values when it first checks for the existance of the MyCategory.txt file.

```powershell
if (-Not (Test-Path -Path $global:myTaskCategory)) {
    Write-Verbose [...]
    Set-Content -Value $script:myTaskDefaultCategories -Path $global:myTaskCategory -Encoding Unicode
}
```

## Thoughts, Opinions, Comments, Sarcasm?
Let me know if there's something I've missed, gotten wrong, or haven't thought about.
Otherwise I'll start working on this on Tuesday (27 Nov 2018).

_ `task -Name 'MyTask Category changes' -Category Personal` _

**Comment by jdhitsolutions** [comment]
That's pretty much it.  I appreciate the help. If you add in a Pester test as well that would be helpful.

**Comment by jdhitsolutions** [comment]
Verified and fixed in v1.8.0. Thanks @shaneis
